index.js

```diff
@@ -648,6 +648,16 @@ window.addEventListener("load", function() {
       } else {
         displayOutcome("motion", "error")("Device Motion is not supported");
       }
+    },
+    "window-close":function(){
+      if(confirm("Confirm before Closing Window?")){
+        addEventListener('beforeunload',function(event){
+          event.preventDefault()
+          return event.returnValue="";
```

Comment:
Since most browsers used to show this string to users, this is probably the most useful place to test UA behaviour. I'd suggest setting a non-empty string, like this:
```suggestion
          return event.returnValue = "[beforeunload test] Do you want to leave this page?";
```